### ISSUE TITLE
Feature Request/Idea: Make Curation Status Labels filterable

### ISSUE THREAD

**Issue by philippconzett** [IQSS/dataverse#9247]
**Overview of the Feature Request**
As a curator, I'd like to be able to filter for datasets according to the Curation Status Label (cf. [https://guides.dataverse.org/en/latest/installation/config.html#allowedcurationlabels](https://guides.dataverse.org/en/latest/installation/config.html#allowedcurationlabels)) they currently are assigned, using the facets on the left-hand side of the Dataverse/collection landing page; cf. the Publication Status section:

![image](https://user-images.githubusercontent.com/21955790/210369201-d6dd9cfb-196f-4c40-a7d2-0b9478e50809.png)

**What kind of user is the feature intended for?**
This feature is intended for Curators, and secondarily for Superusers.

**What inspired the request?**
We are currently testing the use of Curation Status Labels to help us managing curation workflows in our Dataverse installation. Before configuring the labels, we thought that they would be filterable using the facets on the left-hand side of the Dataverse/collection landing page, but testing out the labels, it turned out that this wasn't possible. 

**What existing behavior do you want changed?**
The Curation Status Labels not being filterable.

**Any brand new behavior do you want to add to Dataverse?**
No. Default status labels (e.g., Unpublished, In Review) are already filterable.

**Any related open or closed issues to this feature request?**
Not to my knowledge.

**Comment by philippconzett** [comment]
@adam3smith Just wondering: do you have a workaround for filtering according to Curation Status Labels?

**Comment by adam3smith** [comment]
Yeah, we work off the CSV export with curation labels from  `/api/datasets/listCurationStates?key=<API-KEY>`

**Comment by philippconzett** [comment]
Thanks! Somehow I missed that. Is there a section on this in the API Guide? I couldn't find it.

**Comment by pdurbin** [comment]
Hmm, it probably should have been documented as part of this PR...

- #7967

... but it wasn't. 😞  @philippconzett do you mind creating a dedicated issue for this?

**Comment by pdurbin** [comment]
> Thanks! Somehow I missed that. Is there a section on this in the API Guide? I couldn't find it.

Docs (and some other tweaks) coming here:

- #9556

**Comment by DS-INRA** [comment]
This would be great indeed, is there a SPA parallel for this ? :)

**Comment by pdurbin** [comment]
I've always assumed that some day we'd want to upgrade the curation states to regular search facets. That will make them filterable (and searchable).

I'm not aware of an SPA issue for this, no.

**Comment by cmbz** [comment]
2025/02/10
- During prioritization review...
  - Note that the fields do not currently apply to dataset versions. Meaning that work would need to be done to figure out what to do about Solr search (which is only on published versions). Current search works only on metadata fields; not fields like these.
@philippconzett Please let us know what you think here about this feature request.

**Comment by philippconzett** [comment]
Thanks for the heads-up, @cmbz!

At DataverseNO, we currently handle this through a [table on our info website](https://site.uit.no/dataverseno/admin-en/curatorguide/curation-labels/) (sorry for the look, we'recurrently struggling with our Wordpress theme...), so for us there is no urgent need to get this implemented in the Dataverse software.

I wonder whether this feature could be integrated in a curation management tool like KU Leuven’s review dashboard? @DieuwertjeBloemen 

I'm also pinging @DS-INRAE.

**Comment by DieuwertjeBloemen** [comment]
I'll pick it up with @okaradeniz to see what is possible in the review dashboard. But sounds like an interesting idea to explore.

**Comment by qqmyers** [comment]
FYI: I've just started refactoring curation labels so that they track the creation date and who set them, and keep a history. I'm planning to look into adding a facet as well if I get that far.

**Comment by DS-INRAE** [comment]
Will the history only work during the draft state or allow to know for a published version if the draft had curation labels and which ones?

**Comment by qqmyers** [comment]
Right now the history is only shown for draft versions, but it is not erased from the database. I wasn't sure who should be able to see the history after publication (everyone, those who can view unpublished, admin, superuser?) and where it should go (version table? since each version has it's own curation history), what type of api is needed (per-version getCurationStatus, with the includeHistory flag? another flag for the /listCurationStates? or more granular reports per collection/dataset (since the number of published versions will keep growing) or how to tie that to some form of #10976 (note that the current code sets the curation status back to null as part of publication). If there are clear consensus answers to those types of questions, I could perhaps add it to this PR, or it could be a follow-on PR (possibly for the SPA?).

**Comment by gwendoux** [comment]
It might be interesting to know the workflow that was carried out per version if everything was kept in the database.
It might also be useful to be able to add tags to published versions, especially for curators, in order to note which datasets need to be revised if necessary. (this may have already been suggested by Dimitri in a different issue)
In my opinion, the workflow history should only be accessible to admins, curators and dataset contributors.

**Comment by pdurbin** [comment]
The "workflowcomment" table we added in this PR feels related:

- #3992

That is, we at least store the "reason for return" that the curator writes to the author. This is now supported by the GUI as well as of this PR:

- #10137

It's an example of how we're storing some curation history for a dataset as it approaches publication.
